# This config file is used by ros2_control
controller_manager:
  ros__parameters:
    update_rate: 100  # Hz
    use_sim_time: true
    
    joint_trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController
    
    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

joint_trajectory_controller:
  ros__parameters:
    joints:
      - joint_xy
      - joint_zx
      - joint_needle
      - link_2_aux_joint
      - link_2_joint
      - link_3_aux_joint
      - link_3_joint
      - link_4_aux_joint
      - link_4_joint
      - link_5_aux_joint
      - link_5_joint
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity



